lib/graphql/batch/execution_strategy.rb

```diff
@@ -1,49 +1,43 @@
 module GraphQL::Batch
   class ExecutionStrategy < GraphQL::Query::SerialExecution
-    class OperationResolution < GraphQL::Query::SerialExecution::OperationResolution
-      def result
-        GraphQL::Batch::Promise.resolve(super).sync
-      ensure
-        GraphQL::Batch::Executor.current.clear
-      end
+    def execute(_, _, _)
+      as_promise(super).sync
+    ensure
+      GraphQL::Batch::Executor.current.clear
     end
 
-    class SelectionResolution < GraphQL::Query::SerialExecution::SelectionResolution
-      def result
-        as_promise_unless_resolved(super)
-      end
-
-      private
+    private
 
-      def as_promise_unless_resolved(result)
-        all_promises = []
-        each_promise(result) do |obj, key, promise|
-          obj[key] = nil
-          all_promises << promise.then { |value| obj[key] = value }
+    def as_promise(result)
+      all_promises = []
+      each_promise(result) do |obj, key, promise|
+        obj[key] = nil
+        all_promises << promise.then do |value|
+          obj[key] = value
+          as_promise(result)
```

Comment:
Oops, this should have been `as_promise(result)`.  I guess I should also keep the method as as_promise_unless_resolved so that it doesn't bother returning a promise and depend on a check for whether the promise is pending.